src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -63,18 +63,30 @@ public Command parseCommand(String userInput) throws ParseException {
             return new DeleteCommandParser().parse(arguments);
 
         case ClearCommand.COMMAND_WORD:
+            if (!arguments.isEmpty()) {
+                throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
+            }
             return new ClearCommand();
 
         case FindCommand.COMMAND_WORD:
             return new FindCommandParser().parse(arguments);
 
         case ListCommand.COMMAND_WORD:
+            if (!arguments.isEmpty()) {
+                throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
+            }
             return new ListCommand();
 
         case ExitCommand.COMMAND_WORD:
+            if (!arguments.isEmpty()) {
+                throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
+            }
             return new ExitCommand();
 
         case HelpCommand.COMMAND_WORD:
+            if (!arguments.isEmpty()) {
+                throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
```

Comment:
I think instead of throwing a generic `ParseException(MESSAGE_UNKNOWN_COMMAND)`, should throw `ParseException(HelpCommand.MESSAGE_USAGE)`.  The error can be more specific and help the user recover. Same applies for all the other commands

Reply:
Good point! Updated the implementation.